### PR TITLE
Added Dockerfile and docker-compose.yml

### PR DESCRIPTION
**Pull request by robertvanlienden**

### CODE REVIEW

File: src/Model/MenuItem.php [hunk 1]
```diff
@@ -111,18 +132,19 @@ public function getCMSFields()
         return parent::getCMSFields();
     }
 
-    private function getLinkTypes()
+    private function getLinkTypes(): array
     {
         $linkTypes = self::$link_types;
-        $this->extend('updateLinkTypes', $linkTypes);
+
+        $this->extend('updateLinkTypes', self::$link_types);
```
Comment (robertvanlienden) [inline src/Model/MenuItem.php]: $linkTypes

File: src/Model/MenuItem.php [hunk 2]
```diff
@@ -144,87 +166,75 @@ public function getLink()
...
 
         return parent::canView($member);
     }
 
-    /**
-     * @param null $member
-     * @return bool
-     */
-    public function canEdit($member = null)
+    public function canEdit(mixed $member = null): bool
```
Comment (robertvanlienden) [inline src/Model/MenuItem.php]: int|Member

File: .php-cs-fixer.cache [hunk 3]
```diff
@@ -0,0 +1 @@
...g_import_slash":true,"no_whitespace_in_blank_line":true,"ordered_class_elements":{"order":["use_trait"]},"ordered_imports":{"imports_order":["class","function","const"],"sort_algorithm":"none"},"return_type_declaration":true,"single_blank_line_before_namespace":true,"single_trait_insert_per_statement":true,"ternary_operator_spaces":true,"blank_line_after_namespace":true,"constant_case":true,"elseif":true,"function_declaration":true,"indentation_type":true,"line_ending":true,"lowercase_keywords":true,"no_break_comment":true,"no_closing_tag":true,"no_space_around_double_colon":true,"no_spaces_after_function_name":true,"no_spaces_inside_parenthesis":true,"no_trailing_whitespace":true,"no_trailing_whitespace_in_comment":true,"single_blank_line_at_eof":true,"single_class_element_per_statement":{"elements":["property"]},"single_import_per_statement":true,"single_line_after_imports":true,"switch_case_semicolon_to_colon":true,"switch_case_space":true,"encoding":true,"full_opening_tag":true,"array_push":true,"no_unused_imports":true,"strict_comparison":true,"strict_param":true,"cast_spaces":{"space":"none"}},"hashes":{"src\/Model\/Menu.php":3572943324,"src\/Model\/MenuItem.php":2391585450}}
```
Comment (Dennisprins93) [inline .php-cs-fixer.cache:1]: in gitignore

Comment (robertvanlienden) [reply .php-cs-fixer.cache:1]: Still needs to get remove from Git :wink:

**Pull request merged**